### ISSUE TITLE
OVN crashes when I try to start northd

### ISSUE THREAD

**Issue by spk-hebbar** [ovn-org/ovn#161]
I have installed OVN from source and tried to start northd and ended up with the following crash. Can someone please point out what's going wrong here? 

```
[root@dell-r640-033 ovn]# ovn-ctl start_northd
Starting ovsdb-nb                                          [  OK  ]
EAL: RTE_MBUF_DYNFIELD tailq is already registered
PANIC in tailqinitfn_mbuf_dynfield_tailq():
Cannot initialize tailq: RTE_MBUF_DYNFIELD
6: [ovn-nbctl() [0x42129e]]
5: [/lib64/libc.so.6(__libc_start_main+0x78) [0x7f8f5f9a6d18]]
4: [ovn-nbctl() [0x6252cd]]
3: [ovn-nbctl() [0x418c01]]
2: [ovn-nbctl(__rte_panic+0xba) [0x419184]]
1: [ovn-nbctl(rte_dump_stack+0x1b) [0x60f1eb]]
/usr/local/share/ovn/scripts/ovn-ctl: line 154: 582562 Aborted                 (core dumped) $(echo ovn-${db}ctl | tr _ -) --no-leader-only --db="unix:$sock" init
Starting ovsdb-sb                                          [  OK  ]
EAL: RTE_MBUF_DYNFIELD tailq is already registered
PANIC in tailqinitfn_mbuf_dynfield_tailq():
Cannot initialize tailq: RTE_MBUF_DYNFIELD
6: [ovn-sbctl() [0x4212de]]
5: [/lib64/libc.so.6(__libc_start_main+0x78) [0x7f0e383b2d18]]
4: [ovn-sbctl() [0x5f5e8d]]
3: [ovn-sbctl() [0x418c06]]
2: [ovn-sbctl(__rte_panic+0xba) [0x419189]]
1: [ovn-sbctl(rte_dump_stack+0x1b) [0x5dfdab]]
/usr/local/share/ovn/scripts/ovn-ctl: line 154: 582588 Aborted                 (core dumped) $(echo ovn-${db}ctl | tr _ -) --no-leader-only --db="unix:$sock" init
Starting ovn-northd EAL: RTE_MBUF_DYNFIELD tailq is already registered
PANIC in tailqinitfn_mbuf_dynfield_tailq():
Cannot initialize tailq: RTE_MBUF_DYNFIELD
6: [ovn-northd() [0x422a4e]]
5: [/lib64/libc.so.6(__libc_start_main+0x78) [0x7f44f6d65d18]]
4: [ovn-northd() [0x6451ed]]
3: [ovn-northd() [0x418da0]]
2: [ovn-northd(__rte_panic+0xba) [0x419323]]
1: [ovn-northd(rte_dump_stack+0x1b) [0x62f10b]]
/usr/local/share/openvswitch/scripts/ovs-lib: line 602: 582595 Aborted                 (core dumped) "$@"
                                                           [FAILED]
```

Steps to reproduce:
1. Build and install DPDK v21.11.
```
meson build
cd build; ninja; ninja install; ldconfig
```
2. Build and install OVS with DPDK static libraries.
```
./boot.sh
mkdir _gcc  && cd _gcc 
../configure --with-dpdk=static
make -C _gcc && make -C _gcc install

[root@dell-r640-033 dpdk]# ovs-vsctl get Open_vSwitch . dpdk_version
"DPDK 21.11.0"

[root@dell-r640-033 dpdk]# ovs-vswitchd --version
ovs-vswitchd (Open vSwitch) 3.0.90
DPDK 21.11.0

[root@dell-r640-033 dpdk]# ovs-vsctl --no-wait get Open_vSwitch . other_config
{dpdk-init="true", dpdk-multi-seg-mbufs="true", dpdk-socket-mem="1024,1024", pmd-cpu-mask="0x5000000050000", userspace-tso-enable="false", vlan-limit="0"}
```
3. Build and install OVN with built OVS.
```
./boot.sh
./configure --with-ovs-source=/usr/local/src/ovs/ --with-ovs-build=/usr/local/src/ovs/_gcc
make; make install
```
4. Start ovn northd
`ovn-ctl start_northd`

I'm on the latest commit of OVN - 45f04d88ae8fa6217812abdaa623d66b05fc7b6a

**Comment by numansiddique** [comment]
Please see the reply in the ovs ML.

tl;dr,  please use the OVS submodule present in the OVN source base and don't configure dpdk when configuring OVS.

Thanks

**Comment by spk-hebbar** [comment]
Yes, this worked.
